--- volto-18.35.1-74d82edef/packages/volto/src/components/theme/ContentMetadataTags/ContentMetadataTags.jsx
+++ volto-eea-website-theme/src/customizations/volto/components/theme/ContentMetadataTags/ContentMetadataTags.jsx
@@ -1,8 +1,9 @@
 import React, { useEffect } from 'react';
+import { toPublicURL } from '@plone/volto/helpers/Url/Url';
 import Helmet from '@plone/volto/helpers/Helmet/Helmet';
-import { toPublicURL, getBaseUrl } from '@plone/volto/helpers/Url/Url';
 import { hasApiExpander } from '@plone/volto/helpers/Utils/Utils';
-import { getNavroot } from '@plone/volto/actions/navroot/navroot';
+import { getBaseUrl } from '@plone/volto/helpers/Url/Url';
+import { getNavroot } from '@plone/volto/actions//navroot/navroot';
 import config from '@plone/volto/registry';
 import { useDispatch, useSelector } from 'react-redux';
 
@@ -66,13 +67,23 @@
     const titleAndSiteTitleSeparator =
       config?.settings?.titleAndSiteTitleSeparator || '-';
     const navRootTitle = navroot?.title;
+    const contextIsRoot = navRootTitle === title;
     const siteRootTitle = site?.['plone.site_title'];
-    const titlePart = navRootTitle || siteRootTitle;
+    const parentTitle = !contextIsRoot && props.content?.parent?.title;
 
-    if (includeSiteTitle && titlePart && titlePart !== title) {
-      return seo_title || `${title} ${titleAndSiteTitleSeparator} ${titlePart}`;
+    const baseTitle = seo_title || title;
+    if (includeSiteTitle && siteRootTitle !== title) {
+      const parts = [
+        baseTitle,
+        parentTitle,
+        !contextIsRoot && navRootTitle !== parentTitle && navRootTitle,
+      ].filter((part) => Boolean(part));
+      const titleWithSep = parts.join(` ${titleAndSiteTitleSeparator} `);
+      return siteRootTitle && siteRootTitle !== navRootTitle
+        ? `${titleWithSep} ${siteRootTitle}`
+        : titleWithSep;
     } else {
-      return seo_title || title;
+      return baseTitle;
     }
   };
 
