--- volto-18.35.1-74d82edef/packages/volto/src/components/manage/Diff/DiffField.jsx
+++ volto-eea-website-theme/src/customizations/volto/components/manage/Diff/DiffField.jsx
@@ -12,7 +12,7 @@
 import { Provider } from 'react-intl-redux';
 import { createBrowserHistory } from 'history';
 import { ConnectedRouter } from 'connected-react-router';
-import { useSelector } from 'react-redux';
+import { useSelector, useStore } from 'react-redux';
 import config from '@plone/volto/registry';
 import Api from '@plone/volto/helpers/Api/Api';
 import configureStore from '@plone/volto/store';
@@ -136,6 +136,17 @@
   }
 };
 
+const getWidgetByBehavior = (behavior) =>
+  config.widgets.views.behavior?.[behavior] || null;
+
+const getWidgetByFactory = (factory) =>
+  config.widgets.views.factory?.[factory] || null;
+
+const getDefaultWidget =
+  () =>
+  ({ diff }) =>
+    diff;
+
 /**
  * Diff field component.
  * @function DiffField
@@ -154,6 +165,7 @@
   schema,
   diffLib,
 }) => {
+  const reduxStore = useStore();
   const language = useSelector((state) => state.intl.locale);
   const readable_date_format = {
     dateStyle: 'full',
@@ -165,8 +177,11 @@
       splitWords(String(twoStr)),
     );
   };
-
   let parts, oneArray, twoArray;
+  const Widget =
+    getWidgetByBehavior(schema.behavior) ||
+    getWidgetByFactory(schema.factory) ||
+    getDefaultWidget();
   if (schema.widget) {
     switch (schema.widget) {
       case 'richtext':
@@ -185,7 +200,8 @@
       case 'json': {
         const api = new Api();
         const history = createBrowserHistory();
-        const store = configureStore(window.__data, history, api);
+        const currentState = reduxStore.getState();
+        const store = configureStore(currentState, history, api); //decouple the store from the redux store
         parts = diffWords(
           ReactDOMServer.renderToStaticMarkup(
             <Provider store={store}>
@@ -207,7 +223,8 @@
       case 'slate': {
         const api = new Api();
         const history = createBrowserHistory();
-        const store = configureStore(window.__data, history, api);
+        const currentState = reduxStore.getState();
+        const store = configureStore(currentState, history, api);
         parts = diffWords(
           ReactDOMServer.renderToStaticMarkup(
             <Provider store={store}>
@@ -233,7 +250,8 @@
         if (Widget) {
           const api = new Api();
           const history = createBrowserHistory();
-          const store = configureStore(window.__data, history, api);
+          const currentState = reduxStore.getState();
+          const store = configureStore(currentState, history, api);
           parts = diffWords(
             ReactDOMServer.renderToStaticMarkup(
               <Provider store={store}>
@@ -273,33 +291,51 @@
       {view === 'split' && (
         <Grid.Row>
           <Grid.Column width={6} verticalAlign="top">
-            <span
-              dangerouslySetInnerHTML={{
-                __html: join(
-                  map(parts, (part) => {
-                    let combined = (part.value || []).reduce((acc, value) => {
-                      return acc + formatDiffPart(part, value, 'left');
-                    }, '');
-                    return combined;
-                  }),
-                  '',
-                ),
-              }}
+            <Widget
+              value={one}
+              diff={
+                <span
+                  dangerouslySetInnerHTML={{
+                    __html: join(
+                      map(parts, (part) => {
+                        let combined = (part.value || []).reduce(
+                          (acc, value) => {
+                            return acc + formatDiffPart(part, value, 'left');
+                          },
+                          '',
+                        );
+                        return combined;
+                      }),
+                      '',
+                    ),
+                  }}
+                />
+              }
+              newValue={two}
             />
           </Grid.Column>
           <Grid.Column width={6} verticalAlign="top">
-            <span
-              dangerouslySetInnerHTML={{
-                __html: join(
-                  map(parts, (part) => {
-                    let combined = (part.value || []).reduce((acc, value) => {
-                      return acc + formatDiffPart(part, value, 'right');
-                    }, '');
-                    return combined;
-                  }),
-                  '',
-                ),
-              }}
+            <Widget
+              value={two}
+              diff={
+                <span
+                  dangerouslySetInnerHTML={{
+                    __html: join(
+                      map(parts, (part) => {
+                        let combined = (part.value || []).reduce(
+                          (acc, value) => {
+                            return acc + formatDiffPart(part, value, 'right');
+                          },
+                          '',
+                        );
+                        return combined;
+                      }),
+                      '',
+                    ),
+                  }}
+                />
+              }
+              oldValue={one}
             />
           </Grid.Column>
         </Grid.Row>
@@ -307,18 +343,28 @@
       {view === 'unified' && (
         <Grid.Row>
           <Grid.Column width={16} verticalAlign="top">
-            <span
-              dangerouslySetInnerHTML={{
-                __html: join(
-                  map(parts, (part) => {
-                    let combined = (part.value || []).reduce((acc, value) => {
-                      return acc + formatDiffPart(part, value, 'unified');
-                    }, '');
-                    return combined;
-                  }),
-                  '',
-                ),
-              }}
+            <Widget
+              value={two}
+              one={one}
+              two={two}
+              diff={
+                <span
+                  dangerouslySetInnerHTML={{
+                    __html: join(
+                      map(parts, (part) => {
+                        let combined = (part.value || []).reduce(
+                          (acc, value) => {
+                            return acc + formatDiffPart(part, value, 'unified');
+                          },
+                          '',
+                        );
+                        return combined;
+                      }),
+                      '',
+                    ),
+                  }}
+                />
+              }
             />
           </Grid.Column>
         </Grid.Row>
