--- @plone/volto@18.35.1/components/manage/Blocks/Block/Edit.jsx
+++ src/customizations/volto/components/manage/Blocks/Block/Edit.jsx
@@ -154,14 +154,16 @@
               e.stopPropagation();
               this.props.setUIState({ hovered: null });
             }}
+            onMouseDown={(e) => {
+              const isMultipleSelection = e.shiftKey || e.ctrlKey || e.metaKey;
+              if (!this.props.selected && isMultipleSelection) {
+                e.preventDefault();
+              }
+            }}
             onClick={(e) => {
               const isMultipleSelection = e.shiftKey || e.ctrlKey || e.metaKey;
-              !this.props.selected &&
-                this.props.onSelectBlock(
-                  this.props.id,
-                  this.props.selected ? false : isMultipleSelection,
-                  e,
-                );
+              (!this.props.selected || isMultipleSelection) &&
+                this.props.onSelectBlock(this.props.id, isMultipleSelection, e);
             }}
             onFocus={(e) => {
               const isMultipleSelection = e.shiftKey || e.ctrlKey || e.metaKey;
