--- volto-18.35.1-74d82edef/packages/volto/src/components/manage/UniversalLink/UniversalLink.tsx
+++ volto-eea-website-theme/src/customizations/volto/components/manage/UniversalLink/UniversalLink.jsx
@@ -1,4 +1,11 @@
+/*
+ * UniversalLink
+ * @module components/UniversalLink
+ * Removed noreferrer from rel attribute
+ */
+
 import React from 'react';
+import PropTypes from 'prop-types';
 import { HashLink as Link } from 'react-router-hash-link';
 import { useSelector } from 'react-redux';
 import {
@@ -6,211 +13,150 @@
   isInternalURL,
   URLUtils,
 } from '@plone/volto/helpers/Url/Url';
+
 import config from '@plone/volto/registry';
-import cx from 'classnames';
-import type { ObjectBrowserItem } from '@plone/types';
 
-type BaseProps = {
-  openLinkInNewTab?: boolean;
-  download?: boolean;
-  children: React.ReactNode;
-  className?: string;
-  title?: string;
-  smooth?: boolean;
-  onClick?: (e: React.MouseEvent) => void;
-  onKeyDown?: (e: React.KeyboardEvent) => void;
-};
+const UniversalLink = ({
+  href,
+  item = null,
+  openLinkInNewTab,
+  download = false,
+  children,
+  className = null,
+  title = null,
+  ...props
+}) => {
+  const token = useSelector((state) => state.userSession?.token);
 
-type HrefOnly = {
-  href: string;
-  item?: never;
-} & BaseProps;
+  let url = href;
 
-type ItemOnly = {
-  href?: never;
-  item: Partial<ObjectBrowserItem> & { remoteUrl?: string };
-} & BaseProps;
+  if (!href && item) {
+    if (!item['@id']) {
+      // eslint-disable-next-line no-console
+      console.error(
+        'Invalid item passed to UniversalLink',
+        item,
+        props,
+        children,
+      );
+      url = '#';
+    } else {
+      //case: generic item
+      url = flattenToAppURL(item['@id']);
 
-export type UniversalLinkProps = HrefOnly | ItemOnly;
-export interface AppState {
-  content: {
-    data?: {
-      '@components'?: {
-        translations?: {
-          items?: { language: string; '@id': string }[];
-        };
-      };
-    };
-  };
-  navroot: {
-    data: {
-      navroot: {
-        id: string;
-      };
-    };
-  };
-  intl: {
-    locale: string;
-  };
-  userSession: {
-    token: string | null;
-  };
-}
+      //case: item like a Link
+      let remoteUrl = item.remoteUrl || item.getRemoteUrl;
+      if (!token && remoteUrl) {
+        url = remoteUrl;
+      }
 
-export const __test = {
-  renderCounter: () => {},
-};
+      //case: item of type 'File'
+      if (
+        download &&
+        config.settings.downloadableObjects.includes(item['@type'])
+      ) {
+        url = url.includes('/@@download/file') ? url : `${url}/@@download/file`;
+      }
 
-export function getUrl(
-  props: UniversalLinkProps,
-  token: string | null,
-  item: UniversalLinkProps['item'],
-  children: React.ReactNode,
-): string {
-  if ('href' in props && typeof props.href === 'string') {
-    return props.href;
+      if (
+        !token &&
+        config.settings.viewableInBrowserObjects.includes(item['@type'])
+      ) {
+        url = `${url}/@@display-file/file`;
+      }
+    }
   }
 
-  if (!item || item['@id'] === '') return config.settings.publicURL;
-  if (!item['@id']) {
-    // eslint-disable-next-line no-console
-    console.error(
-      'Invalid item passed to UniversalLink',
-      item,
-      props,
-      children,
-    );
-    return '#';
+  if (download && isInternalURL(url)) {
+    url = url.includes('/@@download/file') ? url : `${url}/@@download/file`;
   }
 
-  let url = flattenToAppURL(item['@id']);
-  const remoteUrl = item.remoteUrl || item.getRemoteUrl;
+  const isExternal = !isInternalURL(url);
+  const isDownload =
+    (!isExternal && url && url.includes('@@download')) || download;
 
-  if (!token && remoteUrl) {
-    url = remoteUrl;
-  }
+  const isDisplayFile =
+    (!isExternal && url.includes('@@display-file')) || false;
+  const checkedURL = URLUtils.checkAndNormalizeUrl(url);
 
-  if (
-    !token &&
-    item['@type'] &&
-    config.settings.downloadableObjects.includes(item['@type'])
-  ) {
-    url = `${url}/@@download/file`;
-  }
+  // we can receive an item with a linkWithHash property set from ObjectBrowserWidget
+  // if so, we use that instead of the url prop
+  url = (item && item['linkWithHash']) || checkedURL.url;
+  let tag = (
+    <Link
+      to={flattenToAppURL(url)}
+      target={openLinkInNewTab ?? false ? '_blank' : null}
+      title={title}
+      className={className}
+      smooth={config.settings.hashLinkSmoothScroll}
+      {...props}
+    >
+      {children}
+    </Link>
+  );
 
-  if (
-    !token &&
-    item['@type'] &&
-    config.settings.viewableInBrowserObjects.includes(item['@type'])
-  ) {
-    url = `${url}/@@display-file/file`;
+  if (isExternal) {
+    tag = (
+      <a
+        href={url}
+        title={title}
+        target={
+          !checkedURL.isMail &&
+          !checkedURL.isTelephone &&
+          !(openLinkInNewTab === false)
+            ? '_blank'
+            : null
+        }
+        rel="noopener"
+        className={className}
+        {...props}
+      >
+        {children}
+      </a>
+    );
+  } else if (isDownload) {
+    tag = (
+      <a
+        href={flattenToAppURL(url)}
+        download
+        title={title}
+        className={className}
+        {...props}
+      >
+        {children}
+      </a>
+    );
+  } else if (isDisplayFile) {
+    tag = (
+      <a
+        href={flattenToAppURL(url)}
+        title={title}
+        target={!(openLinkInNewTab === false) ? '_blank' : null}
+        rel="noopener"
+        className={className}
+        {...props}
+      >
+        {children}
+      </a>
+    );
   }
+  return tag;
+};
 
-  return url;
-}
+UniversalLink.propTypes = {
+  href: PropTypes.string,
+  openLinkInNewTab: PropTypes.bool,
+  download: PropTypes.bool,
+  className: PropTypes.string,
+  title: PropTypes.string,
+  item: PropTypes.shape({
+    '@id': PropTypes.string.isRequired,
+    remoteUrl: PropTypes.string, //of plone @type 'Link'
+  }),
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node,
+  ]),
+};
 
-const UniversalLink = React.memo(
-  React.forwardRef<HTMLAnchorElement | HTMLDivElement, UniversalLinkProps>(
-    function UniversalLink(props, ref) {
-      const {
-        openLinkInNewTab,
-        download,
-        children,
-        className,
-        title,
-        smooth,
-        onClick,
-        onKeyDown,
-        item,
-        href,
-        ...rest
-      } = props;
-      __test.renderCounter();
-
-      const token = useSelector<AppState, string | null>(
-        (state) => state.userSession?.token,
-      );
-
-      let url = getUrl(props, token, item, children);
-
-      const isExternal = !isInternalURL(url);
-
-      const isDownload =
-        (!isExternal && url.includes('@@download')) || download;
-      const isDisplayFile =
-        (!isExternal && url.includes('@@display-file')) || false;
-
-      const checkedURL = URLUtils.checkAndNormalizeUrl(url);
-
-      url = checkedURL.url;
-      let tag = (
-        <Link
-          to={flattenToAppURL(url)}
-          target={openLinkInNewTab ?? false ? '_blank' : undefined}
-          title={title}
-          className={className}
-          onClick={onClick}
-          smooth={smooth ?? config.settings.hashLinkSmoothScroll}
-          // @ts-ignore
-          ref={ref}
-          {...rest}
-        >
-          {children}
-        </Link>
-      );
-
-      if (isExternal) {
-        const isTelephoneOrMail = checkedURL.isMail || checkedURL.isTelephone;
-        const getClassName = cx({ external: !isTelephoneOrMail }, className);
-
-        tag = (
-          <a
-            href={url}
-            title={title}
-            target={
-              !isTelephoneOrMail && !(openLinkInNewTab === false)
-                ? '_blank'
-                : undefined
-            }
-            rel="noopener noreferrer"
-            {...rest}
-            className={getClassName}
-            ref={ref as React.Ref<HTMLAnchorElement>}
-          >
-            {children}
-          </a>
-        );
-      } else if (isDownload) {
-        tag = (
-          <a
-            href={flattenToAppURL(url)}
-            download
-            title={title}
-            {...rest}
-            className={className}
-            ref={ref as React.Ref<HTMLAnchorElement>}
-          >
-            {children}
-          </a>
-        );
-      } else if (isDisplayFile) {
-        tag = (
-          <a
-            title={title}
-            target="_blank"
-            rel="noopener noreferrer"
-            {...rest}
-            href={flattenToAppURL(url)}
-            className={className}
-            ref={ref as React.Ref<HTMLAnchorElement>}
-          >
-            {children}
-          </a>
-        );
-      }
-      return tag;
-    },
-  ),
-);
-
 export default UniversalLink;
