--- volto-18.35.1-74d82edef/packages/volto/src/components/manage/Sharing/Sharing.jsx
+++ volto-eea-website-theme/src/customizations/volto/components/manage/Sharing/Sharing.jsx
@@ -554,7 +554,7 @@
       available_roles: state.sharing.data.available_roles,
       updateRequest: state.sharing.update,
       pathname: props.location.pathname,
-      title: state.content.data.title,
+      title: state.content.data?.title || '',
       login: state.userSession.token
         ? jwtDecode(state.userSession.token).sub
         : '',
